src/h3lib/lib/algos.c

```diff
@@ -813,6 +817,58 @@ H3Error _getEdgeHexagons(const GeoLoop *geoloop, int64_t numHexagons, int res,
     return E_SUCCESS;
 }
 
+/**
+ * Returns true if the given cell is in the polygon using the containment
+ * mode flags.
+ */
+bool cellInPolygon(H3Index hex, const GeoPolygon *geoPolygon,
```

Comment:
Per discussion, this adds a few modes we know we can implement (ANY_VERTEX, ALL_VERTEXES) which shouldn't require any more complicated changes to the library right now.